code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -230,26 +236,43 @@ func (mrh *MinerRestHandler) getNodePoolStat(w http.ResponseWriter, r *http.Requ
 	var (
 		id     = r.URL.Query().Get("id")
 		poolID = r.URL.Query().Get("pool_id")
-		status = r.URL.Query().Get("status")
-		sn     *MinerNode
 		err    error
 	)
 
-	if sn, err = getMinerNode(id, mrh.GetQueryStateContext()); err != nil {
-		common.Respond(w, r, nil, sc.NewErrNoResourceOrErrInternal(err, true, "can't get miner node"))
+	edb := mrh.GetQueryStateContext().GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
 		return
 	}
 
-	if poolID == "" {
-		common.Respond(w, r, sn.GetNodePools(status), nil)
+	_, err = edb.GetMiner(id)
```

Comment:
do we need this check?

Reply:
It seems to be a routine check. If a miner doesn't exist, it is probably not supposed to have have node pool.

Reply:
remove it please, since it is not important additional request to db, we won't have pool anyway, so we will have error, but with slightly different text

Reply:
Removed this, apparenty this endpoint is used for both miner and sharder for system test, (even if the command is mn-pool-info), so querying nodePollStat for sharder produces unexpected error. SO removing it was the only solution.

Reply:
Also removed providerType parameter from query so it works for both sharder and miner